packages/salesforcedx-vscode-core/esbuild.config.js

```diff
@@ -15,35 +16,40 @@ const sharedConfig = {
   external: [
     'vscode',
     'applicationinsights',
-    'shelljs',
-    '@salesforce/core-bundle',
-    '@salesforce/templates-bundle'
+    'shelljs'
   ],
-  minify: true
+  minify: true,
+  keepNames: true,
+  plugins: [
+    esbuildPluginPino({ transports: ['pino-pretty'] })
+  ]
+};
+
+const copyFiles = async (src, dest) => {
+  try {
+    await fs.copy(src, dest);
+    console.log(`Copied from ${src} to ${dest}`);
+  } catch (error) {
+    console.error('An error occurred:', error);
+  }
 };
 
 // copy core-bundle/lib/transformStream.js to dist if core-bundle is included
-// const copyFiles = async (src, dest) => {
-//   try {
-//     // Copy the file
-//     await fs.copyFile(src, dest);
-//     console.log(`File was copied from ${src} to ${dest}`);
-//   } catch (error) {
-//     console.error('An error occurred:', error);
-//   }
-// };
+const srcPathTransformStream = '../../node_modules/@salesforce/core-bundle/lib/transformStream.js';
+const destPathTransformStream = './dist/src/transformStream.js';
 
-// const srcPath = '../../node_modules/@salesforce/core-bundle/lib/transformStream.js';
-// const destPath = './dist/transformStream.js';
+const srcTemplatesPath = '../../node_modules/@salesforce/templates/lib/templates';
+const destTemplatesPath = './dist/templates';
```

Comment:
I made the structure bundle artifacts into a /src folder and a /templates folder, so when /dist/src/index.js refers to the generators from templates, the path to templates folder is identical to the path from [baseGenerator](https://github.com/forcedotcom/salesforcedx-templates/blob/main/src/generators/baseGenerator.ts#L86) to /templates in the original library.

Reply:
@mingxuanzhangsfdx could we get the constraints of coordinating file placement within a bundle and the corresponding needs documented some where?

Reply:
Having these mappings bothers me because these are things that need to be remembered.

Given we are consuming templates in this way, what can we do to templates (and other modules that need to be bundled) so we retain structure moving a->b?